README.md

```diff
@@ -154,25 +152,30 @@ Here is a good editor for regexp, maybe try and do it without using it first tho
 ```ruby
 # /User/gollum/precious/ring.rb
 File.dirname(__FILE__)
+
+File.dirname("/User/gollum/precious/")
 ```
 
-- *give me answer*
 
 ```ruby
 # /User/tom/junk/ring.rb
 File.path(__dir__)
+
+File.path("/User/tom/junk/ring.rb")
 ```
 
-- *give me answer*
 
 #### Write the following method that would give me the correct return value:
 
 ```ruby
 bark { "Oki" }
```

Comment:
Here we are passing a block to the `bark` method, there are two ways we could write this:
```ruby
# Not &block is just a name but I've seen it used a lot, you could call it anything ex `&dogs_name`
def bark(&block)
  puts "#{block.call} yips!"
end

# alternate way
def bark
  puts "#{yield} yips!"
end
```
You can also pass arguments to a block:
```ruby
bark { |emotion| "#{emotion} Oki" }
# => Sad Oki whines!
# => Happy Oki yips!

def bark(&block)
  puts "#{block.call("Sad")} whines!"
  puts "#{block.call("Happy")} yips!"
end

def bark
  puts "#{yield("Sad")} whines!"
  puts "#{yield("Happy")} yips!"
end
```

Reply:
I did a quick search, looks like this is an alright article to give some more details on blocks procs and stuff. Might be better articles tho. https://www.rubyguides.com/2016/02/ruby-procs-and-lambdas/